project-management/src/main/java/life/qbic/projectmanagement/application/api/AsyncProjectService.java

```diff
@@ -93,6 +98,58 @@ Mono<ProjectCreationResponse> create(ProjectCreationRequest request)
       throws UnknownRequestException, RequestFailedException, AccessDeniedException;
 
 
+  /**
+   * Requests {@link SamplePreview} for a given experiment.
+   *
+   * @param experimentId the experiment ID for which the sample preview shall be retrieved
+   * @return a reactive stream of {@link SamplePreview} objects of the experiment
+   * @throws RequestFailedException if the request could not be executed
+   * @since 1.10.0
+   */
+  Flux<SamplePreview> getSamplePreviews(String experimentId) throws RequestFailedException;
+
+  /**
+   * Requests {@link SamplePreview} for a given experiment with pagination support.
+   *
+   * @param experimentId the experiment ID for which the sample preview shall be retrieved
+   * @param offset       the offset from 0 of all available previews the returned previews should
+   *                     start
+   * @param limit        the maximum number of previews that should be returned
+   * @return a reactive stream of {@link SamplePreview} objects in the experiment
+   * @since 1.10.0
+   */
+  Flux<SamplePreview> getSamplePreviews(String experimentId, int offset, int limit);
+
+  /**
+   * Requests all {@link Sample} for a given experiment.
+   *
+   * @param experimentId the experiment ID for which the samples shall be retrieved
+   * @return a reactive stream of {@link Sample} objects
+   * @throws RequestFailedException in case the request cannot be executed
+   * @since 1.10.0
+   */
+  Flux<Sample> getSamples(String experimentId) throws RequestFailedException;
+
+  /**
+   * Requests all {@link Sample} for a given batch
+   *
+   * @param batchId the batch ID the samples shall be retrieved for
+   * @return a reactive stream of {@link Sample} objects for the given batch
+   * @throws RequestFailedException in case the request cannot be executed
+   * @since 1.10.0
+   */
+  Flux<Sample> getSamplesForBatch(String batchId) throws RequestFailedException;
```

Comment:
Do we want to still expose the domain model sample here? Tracking which information we actually use later on is part of coming up with a good API.

Reply:
Does this actually throw an exception or does the Flux have an error element?